lib/vm/src/memory.rs

```diff
@@ -379,9 +421,15 @@ impl LinearMemory for VMSharedMemory {
         guard.vm_memory_definition.as_ptr()
     }
 
-    /// Owned memory can not be cloned (this will always return None)
+    /// Shared memory can always be cloned
     fn try_clone(&self) -> Option<Box<dyn LinearMemory + 'static>> {
-        None
+        Some(Box::new(self.clone()))
+    }
+
+    /// Copies this memory to a new memory
+    fn fork(&mut self) -> Result<Box<dyn LinearMemory + 'static>, MemoryError> {
```

Comment:
Let's use clone. I think is a more accurate naming

Reply:
Renamed to `duplicate()`.